### PR TITLE
Upgrade Project to React v19

### PR DESCRIPTION
**Pull request by smilkuri**
Issue

Fixes: https://github.com/aws-samples/aws-sdk-js-notes-app/issues/155

Description
This PR updates the application to React v19 by bumping the React and related module versions.

How was this change tested?
Local

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,5 +27,13 @@
...
-  "packageManager": "yarn@4.4.0"
+  "packageManager": "yarn@4.4.0",
+  "dependencies": {
+    "@types/react": "^19.0.0",
+    "@types/react-dom": "^19.0.0",
+    "bootstrap": "^5.3.3",
+    "react": "19.0.0",
+    "react-bootstrap": "^2.10.7",
+    "react-dom": "19.0.0"
+  }
```
Comment (trivikr) [inline package.json:38]: These dependencies are not required in the base package.json of monorepo.
```suggestion
  "packageManager": "yarn@4.4.0"
```

They're added in `packages/frontend/package.json`

File: package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
     "start:frontend": "cd packages/frontend && yarn start"
   },
   "devDependencies": {
+    "aws-cdk": "^2.173.2",
```
Comment (trivikr) [inline package.json:16]: The "aws-cdk" dependency is also not required in root package.json. It's present in `infra/package.json`
```suggestion
```

File: packages/frontend/package.json [hunk 3]
```diff
@@ -13,11 +13,12 @@
     "@aws-sdk/s3-request-presigner": "3.627.0",
     "@aws-sdk/util-create-request": "3.622.0",
     "@aws-sdk/util-format-url": "3.609.0",
-    "@reach/router": "1.3.4",
+    "@reach/router": "^1.3.4",
```
Comment (trivikr) [inline packages/frontend/package.json:16]: I don't think `@reach/router` is required to be updated to bump React version.
```suggestion
    "@reach/router": "1.3.4",
```

It's latest version is `1.3.4` https://www.npmjs.com/package/@reach/router?activeTab=versions, as it's deprecated in favor of React Router.
That migration is tracked in https://github.com/aws-samples/aws-sdk-js-notes-app/issues/26

File: packages/frontend/package.json [hunk 4]
```diff
@@ -13,11 +13,12 @@
...
     "p-event": "^6.0.1",
-    "react": "18.2.0",
+    "react": "^19.0.0",
```
Comment (trivikr) [inline packages/frontend/package.json:18]: Use fixed version of react dependencies.
```suggestion
    "react": "19.0.0",
```

When users try code from aws-samples, we want them to use a working version.

File: packages/frontend/package.json [hunk 5]
```diff
@@ -13,11 +13,12 @@
...
     "react-bootstrap": "1.4.0",
-    "react-dom": "18.2.0"
+    "react-dom": "^19.0.0",
+    "react-router-dom": "^7.1.1"
```
Comment (trivikr) [inline packages/frontend/package.json:21]: Why is react-router-dom needed for React v19 upgrade? We use reach router.

File: packages/frontend/package.json [hunk 6]
```diff
@@ -55,11 +56,11 @@
   "devDependencies": {
     "@tsconfig/node20": "^20.1.4",
     "@types/node": "^18.11.18",
-    "@types/reach__router": "1.3.6",
+    "@types/reach__router": "^1",
     "@types/react": "^18.0.15",
     "@types/react-dom": "^18.0.6",
     "@vitejs/plugin-react-refresh": "1.3.6",
-    "react-bootstrap-icons": "1.3.0",
+    "react-bootstrap-icons": "^1.11.5",
```
Comment (trivikr) [inline packages/frontend/package.json:63]: Is update to react-bootstrap-icons required to bump to React v19?
```suggestion
    "react-bootstrap-icons": "1.11.5",
```

If yes, use fixed version

**Review by trivikr — CHANGES_REQUESTED** [review]

File: packages/frontend/package.json (hunk 6 above)
Comment (smilkuri) [reply packages/frontend/package.json]: Not necessarily. Just updated to keep it up-to-date. But older version also works.

File: packages/frontend/package.json (hunk 5 above)
Comment (smilkuri) [reply packages/frontend/package.json]: Yes, modified it.

**Review by trivikr — APPROVED** [review]

**Pull request merged**